.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,95 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest,ubuntu-latest,macos-12]
+        include:
+          - os: ubuntu-latest
+            python-version: '3.12'
+            toolchain: {compiler: gcc, version: 13}
+          - os: macos-12
+            python-version: '3.12'
+            toolchain: {compiler: gcc, version: 13}
+          - os: windows-latest
+            python-version: '3.11'
+            toolchain: {compiler: gcc, version: 13}
+            
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Fortran compiler for Unix
+        if: runner.os != 'Windows'
+        uses: awvwgk/setup-fortran@v1
+        with:
+          compiler: ${{ matrix.toolchain.compiler }}
+          version: ${{ matrix.toolchain.version }}
+
+      - name: Install Fortran compiler on Windows
+        if: runner.os == 'Windows'
+        run: |
+          Set-ExecutionPolicy Bypass -Scope Process -Force
+          [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072
+          iex ((New-Object System.Net.WebClient).DownloadString('https://community.chocolatey.org/install.ps1'))
+          choco install gcc
+          choco install make
+        shell: pwsh
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Build wheels
```

Comment:
I am sure there is a way to build directly from pip. However,  I had difficulty building from pip, since setup requires compiling fortran files using 'f2py'. I found using cibuildwheel makes .whl build much easier without any error.